skins/oasis/modules/BodyController.class.php

```diff
@@ -77,286 +111,309 @@ public static function isGridLayoutEnabled() {
+				if ( $wg->EnableHuluVideoPanel ) {
+					$railModuleList[$huluVideoPanelKey] = [ 'HuluVideoPanel', 'Index', null ];
 				}
 			}
 		}
 
 		// User page namespaces
-		if( in_array($wgTitle->getNamespace(), self::getUserPagesNamespaces() ) ) {
-			$page_owner = User::newFromName($wgTitle->getText());
+		if ( in_array( $wg->Title->getNamespace(), self::getUserPagesNamespaces() ) ) {
+			$page_owner = User::newFromName( $wg->Title->getText() );
 
-			if($page_owner) {
-				if ( !$page_owner->getGlobalPreference('hidefollowedpages') ) {
-					$railModuleList[1101] = array('FollowedPages', 'Index', null);
+			if ( $page_owner ) {
+				if ( !$page_owner->getGlobalPreference( 'hidefollowedpages' ) ) {
+					$railModuleList[1101] = [ 'FollowedPages', 'Index', null ];
 				}
 
-				if ( $wgEnableAchievementsExt ) {
-					$railModuleList[1102] = array('Achievements', 'Index', null);
+				if ( $wg->EnableAchievementsExt ) {
+					$railModuleList[1102] = [ 'Achievements', 'Index', null ];
 				}
 			}
 		}
 
-		if (self::isBlogPost() || self::isBlogListing()) {
-			$railModuleList[1250] = array('PopularBlogPosts', 'Index', null);
+		if ( self::isBlogPost() || self::isBlogListing() ) {
+			$railModuleList[1250] = [ 'PopularBlogPosts', 'Index', null ];
 		}
 
 		//  No rail on main page or edit page for oasis skin
 		// except &action=history of wall
-		if( !empty($wgEnableWallEngine) ) {
-			$isEditPage = !WallHelper::isWallNamespace($namespace) && BodyController::isEditPage() || $wgRequest->getVal('diff');
+		if ( !empty( $wg->EnableWallEngine ) ) {
+			$isEditPage = !WallHelper::isWallNamespace( $namespace ) && BodyController::isEditPage() || $wg->Request->getVal('diff');
 		} else {
 			$isEditPage = BodyController::isEditPage();
 		}
 
 		if ( $isEditPage || WikiaPageType::isMainPage() ) {
-			$modules = array();
-			wfRunHooks( 'GetEditPageRailModuleList', array( &$modules ) );
+			$modules = [];
+			wfRunHooks( 'GetEditPageRailModuleList', [ &$modules ] );
 			wfProfileOut(__METHOD__);
 			return $modules;
 		}
+
 		// No modules on Custom namespaces, unless they are in the ContentNamespaces list, those get the content rail
-		if (is_array($wgExtraNamespacesLocal) && array_key_exists($subjectNamespace, $wgExtraNamespacesLocal) && !in_array($subjectNamespace, $wgContentNamespaces)) {
-			wfProfileOut(__METHOD__);
-			return array();
-		}
-		// If the entire page is non readable due to permissions, don't display the rail either RT#75600
-		if (!$wgTitle->userCan( 'read' )) {
-			wfProfileOut(__METHOD__);
-			return array();
+		if (
+			is_array( $wg->ExtraNamespacesLocal ) &&
+			array_key_exists( $subjectNamespace, $wg->ExtraNamespacesLocal ) &&
+			!in_array( $subjectNamespace, $wg->ContentNamespaces )
+		) {
+			return [];
 		}
 
-		$railModuleList[1440] = array('Ad', 'Index', ['slotName' => 'TOP_RIGHT_BOXAD']);
-		$railModuleList[1435] = array('AdEmptyContainer', 'Index', ['slotName' => 'NATIVE_TABOOLA_RAIL']);
```

Comment:
Let's bring it back, please. We'll probably remove `NATIVE_TABOOLA_RAIL` one day but `TOP_RIGHT_BOXAD` is quite important for the company :smiley_cat: